pkg/scheduler/scheduler.go

```diff
@@ -124,6 +154,9 @@ func (s *Scheduler) schedule(ctx context.Context) {
 		log := log.WithValues("workload", klog.KObj(e.Obj), "clusterQueue", klog.KRef("", e.ClusterQueue))
 		if err := s.admit(ctrl.LoggerInto(ctx, log), e); err != nil {
```

Comment:
The reasoning to block early in this function was that the blocking may take a while (say a couple of minutes) and after the time the nominated workloads could be different. However, this seems a minor concern. 

I think that another advantage of the suggested approach is that this way it will be easier to implement the more granular version of the feature with blocking based on overlapping resource flavors. Going to follow up with a commit along these lines.

Reply:
You make a good point. Workloads could have finished and freed resources. But at least we know that this workload would fit.

Reply:
Moved the waiting just before admit

Reply:
> Also, it might be useful to produce a Workload status update if the Workload satisfies the quota, but is blocked waiting on others Workloads to be ready.

Now, I set the `WorkloadAdmitted=False` condition with `Reason=Waiting`. It allows me to wait for the situation in the integration test.